src/xtgeo/surface/_regsurf_gridding.py

```diff
@@ -39,46 +162,1551 @@ def points_gridding(self, points, method="linear", coarsen=1):
+                        znew[i0, j1] += w01 * z_adjust
+                        znew[i1, j1] += w11 * z_adjust
+
+                        n_enforced += 1
+                        break
+
+                if found_cell:
+                    break
+
+        if not found_cell:
+            n_outside += 1
+
+    logger.debug(
+        "Enforced exact values for %d/%d points using bilinear adjustment "
+        "(%d points outside grid)",
+        n_enforced,
+        n_tested,
+        n_outside,
+    )
+    return n_enforced
+
+
+def points_kriging_gridding(
+    self: RegularSurface,
+    points: Points,
+    variogram_model: str = "Gaussian",
+    variogram_parameters: dict[str, Any] | None = None,
+    len_scale: float | tuple[float, float] | None = None,
+    range_value: float | tuple[float, float] | None = None,
+    nugget: float = 0.0,
+    krige_type: Literal["ordinary", "simple"] = "ordinary",
+    mean: float | None = None,
+    coarsen: int = 1,
+    max_points: int = 500,
+    radius: float | None = None,
+    exact: bool = False,
+    angle: float | None = None,
+) -> None:
+    """
+    Do kriging gridding from a points data set using GSTools.
+
+    Kriging provides optimal spatial interpolation based on spatial correlation
+    structure (variogram). Good for sparse, spatially correlated data.
+
+    Args:
+        points: Points object containing x, y, z coordinates
+        variogram_model: Variogram/Covariance model. Options:
+            - 'Gaussian': Gaussian model (smooth, good for continuous phenomena)
+            - 'Exponential': Exponential model (good for geology)
+            - 'Spherical': Spherical model (classic geostatistics)
+            - 'Matern': Matérn model (flexible smoothness)
+            - 'Stable': Stable model (generalization of Gaussian, also called
+              'General exponential')
+            - 'Linear': Linear model
+        len_scale: Correlation length (range). If None, auto-estimated from data.
+            Can be single value or tuple (len_scale_x, len_scale_y) for anisotropy.
+            First value is major axis (X or Easting), second is minor axis (Y or
+            Northing).
+        range_value: Alternative to len_scale. If provided, converted to len_scale via
+            variogram model conventions (e.g. RMS/Petrel style). Ignored if len_scale
+            is given. Can be single value or tuple for anisotropy. First value is
+            major axis (X or Easting), second is minor axis (Y or Northing).
+        nugget: Nugget effect (measurement error/micro-scale variation), default 0.0
+        krige_type: Type of kriging, either 'ordinary' (default) or 'simple'
+        mean: Mean value for simple kriging. If None and krige_type='simple',
+            the mean will be estimated from the data. Not used for ordinary kriging.
+        variogram_parameters: Dict with additional model parameters:
+            - 'var' or 'sill': variance/sill of the model
```

Comment:
Why the two options for setting the variance? I'd prefer to have this as "variance" instead which is more explicit.
```suggestion
            - 'variance': variance (sill) of the model
```